GitUI/BuildServerIntegration/BuildServerWatcher.cs

```diff
@@ -200,6 +203,30 @@ public IBuildServerCredentials GetBuildServerCredentials(IBuildServerAdapter bui
             }
         }
 
+        public string ReplaceVariables(string projects)
```

Comment:
used projectNames, same as the callers use

Reply:
I'm still struggling to understand its purpose. Can you provide an example of a value the parameter carries?

Reply:
"projectNames" has the configuration of the build servers, for GE I have "gitextensions/%REPO_SHORTNAME%|gerhardol/%REPO_SHORTNAME%' to see my private bulds as well as the "official".
(The example indicates that "REPO_USERNAME% or something could be useful too, but I cannot explain it good enough)

Reply:
Thanks, it is clearer now.
Could you please add a comment or xml-doc describing this.

Reply:
perhaps it can be refactored as follows:

```
public interface IBuildServerProjectNameFormatter
{
    /// xml-doc goes here explaining the purpose
    string ReplaceVariables(string projectNames);
}

public sealed class BuildServerProjectNameFormatter : IBuildServerProjectNameFormatter
{
    public BuildServerProjectNameFormatter(Func<IGitModule> getModule)
    {
        // set the dependency the same way we do in other places
    }

    public string ReplaceVariables(string projectNames)
    {
        // the implementation goes here
    }
}
```

And the usage merely changes to:

```
             if (!useAllProjets)
             {
                projectNames = _buildServerProjectNameFormatter.ReplaceVariables(projectNamesSetting)
                    .Split(new[] {'|'}, StringSplitOptions.RemoveEmptyEntries);
             }
              if (Projects.Count == 0 ||
                  (!useAllProjets && Projects.Keys.Intersect(projectNames).Count() != projectNames.Length))
              {
```


I am not totally happy with the method name, especially if we want to add an overload which splits the output (so that callers don't have to do it), but can't think of a good name....


```
public interface IBuildServerProjectNameBuilder
{
    string Build(string projectNames);
    string[] Build(string projectNames);
}
```

Maybe?

Reply:
I would prefer to not have the obstacle to have shared code for the plugins and see no benefit in distributing an additional IBuildServerProjectNameFormatter. Suggest to keep ReplaceVariables() in IBuildServerWatcher. 
But GetRepoShortname() can be in a separate class.
Just adding description right now